R/data-repository.R

```diff
@@ -0,0 +1,28 @@
+#' @title DataRepository
+#' @docType class
+#' @description  An object typically holding observed data
+#' @format NULL
+DataRepository <- R6::R6Class(
+  "DataRepository",
+  inherit = DotNetWrapper,
+  cloneable = FALSE,
+  active = list(
+    #' @field baseGrid Returns the base column for the population (typically time column)
+    baseGrid = function(value) {
+      DataColumn$new(private$wrapReadOnlyProperty("BaseGrid", value))
+    },
+    #' @field columns Returns all columns (including baseGrid defined in the data)
+    columns = function(value){

```

Comment:
@PavelBal @Yuri05 I will cache this so that we always return the same instances of DataColumn wrapper